src/Razor/src/Microsoft.AspNetCore.Razor.Common/Telemetry/TelemetryReporter.cs

```diff
@@ -78,6 +96,16 @@ public void ReportFault(Exception exception, string? message, object[] @params)
                 exceptionObject: exception,
                 gatherEventDetails: faultUtility =>
                 {
+                    foreach (var data in @params)
+                    {
+                        if (data is null)
+                        {
+                            continue;
+                        }
+
+                        faultUtility.AddErrorInformation(data.ToString());
```

Comment:
whats a `faultUtility` and what does it do?

Reply:
This is part of the Fault API we consume from VS